charts/influxdb/values.yaml

```
@@ -240,22 +239,56 @@ initScripts:
     init.iql: |+
       CREATE DATABASE "telegraf" WITH DURATION 30d REPLICATION 1 NAME "rp_30d"
 
+
+## InfluxDB backup parameters
+##
 backup:
   enabled: false
-  schedule: "0 0 * * *"
-  annotations: {}
-  podAnnotations: {}
-
-  ## Google Cloud Storage
-  # gcs:
-  #    serviceAccountSecret: influxdb-backup-key
-  #    serviceAccountSecretKey: key.json
-  #    destination: gs://bucket/influxdb
-
-  ## Azure
-  ## Secret is expected to have connection string stored in `connection-string` field
-  ## Existing container will be used or private one withing storage account will be created.
-  # azure:
-  #   storageAccountSecret: influxdb-backup-azure-key
-  #   destination_container: influxdb-container
-  #   destination_path: ""
+  ## Directory where backups are stored
+  ##
+  directory: "/backups"
+  ## Retention time in days for backups
+  ##
+  retentionDays: 10
+  ## Cronjob configuration
+  ## This cronjob is used to create InfluxDB backups
+  ##
+  cronjob:
+    ## Schedule in Cron format to save snapshots
+    ## See https://en.wikipedia.org/wiki/Cron
+    ##
+    schedule: "0 2 * * *"
+    ## Number of successful finished jobs to retain
+    ##
+    historyLimit: 3
+    ## Pod annotations
+    ## ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/
+    ##
+    podAnnotations: {}
+  ## Storage providers where to upload backups
+  ##
+  uploadProviders:
+    ## Google Storage Bucket configuration
+    ##
+    google:
+      enabled: false
+      secret: ""
+      secretKey: "key.json"
+      existingSecret: ""
+      bucketName: "gs://bucket/influxdb"
+      image:
+        registry: docker.io
+        repository: google/cloud-sdk
+        tag: 291.0.0-alpine
+    ## Azure Storage Container configuration
+    ##
+    azure:
+      enabled: false
+      secret: ""
+      secretKey: "connection-string"
+      existingSecret: ""
+      containerName: "influxdb-container"
+      image:
+        registry: docker.io
+        repository: microsoft/azure-cli
+        tag: 2.0.24
```

Please bring back 1 level up and revert the key names (gcs) as to avoid breaking changes.
```suggestion
    google:
      enabled: false
      secret: ""
      secretKey: "key.json"
      existingSecret: ""
      bucketName: "gs://bucket/influxdb"
      image:
        registry: docker.io
        repository: google/cloud-sdk
        tag: 291.0.0-alpine
    ## Azure Storage Container configuration
    ##
    azure:
      enabled: false
      secret: ""
      secretKey: "connection-string"
      existingSecret: ""
      containerName: "influxdb-container"
      image:
        registry: docker.io
        repository: microsoft/azure-cli
        tag: 2.0.24
```

---

i've updated major chart version and added update docu to the readme so no problem to introduce some changes.

---

These are breaking changes that are purely stylistic though, so not worth it imho

---

done

---

But even with 1 level up the changes are breaking.
Imho the renaming is a improvement to the current one as they are much more consistent over the storageproviders.

---

In which case is it more consistent?

bucketname was intended to be destination url as to specify a "folder" in the bucket, why change this?

imho these stylistic changes are not worth breaking the chart for users using this in production today.